### PR TITLE
#1653057257 Add Pagination to Articles

### PR DESCRIPTION
**Pull request by Amoswachira**
#### What does this PR do?
* Adds Pagination feature to articles
#### Description of Task to be completed?
* A user can be able to go to the next page and previous pages of articles
#### How should this be manually tested?
* After cloning the repo:
```
$ git checkout ft-add-pagination-165305725
$ npm install
$ npm run start:dev
```
#### What are the relevant pivotal tracker stories?
[#165305725](https://www.pivotaltracker.com/n/projects/2326460/stories/165305725)
#### Screenshots (if appropriate)
Screenshot 2019-06-20 at 17 52 27

### CODE REVIEW

**Review by IssaIan — APPROVED** [review]

File: src/views/Pagination/PaginationContainer.jsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+                  role="presentation"
+                >
+                  <span aria-hidden="true">&laquo;</span>
+                  <span className="sr-only">Previous</span>
+                </span>
+              </li>
+            ) : null}
+
+            {pageNumbers.map((num, i) => (
+              <li key={i} className="page-item">
```
Comment (Ogutu-Brian) [inline src/views/Pagination/PaginationContainer.jsx]: Hello @Amoswachira ,
It is not advisable to use array indices as keys in lists. You can use the index in combination with other values like we had done before when fetching all articles.

File: src/views/Articles/Posts.jsx [hunk 2]
```diff
@@ -1,73 +1,124 @@
...
+import { getPages, getNextPages } from '../../redux/actions/postActions';
+import articlesUrl from '../../services/api';
 import '../../styles/App.css';
+import '../../styles/pagination.css';
+import PaginationContainer from '../Pagination/PaginationContainer';
 
 
 
 class Posts extends Component {
+  state = {
```
Comment (Ogutu-Brian) [inline src/views/Articles/Posts.jsx:16]: Hello,
You should also ensure that the state is not set within component Did mount

**Review by Ogutu-Brian — CHANGES_REQUESTED** [review]

**Review by IssaIan — APPROVED** [review]

File: src/views/Articles/Posts.jsx (hunk 2 above)
Comment (IssaIan) [reply src/views/Articles/Posts.jsx:16]: The issue is with setting state in the componentDidMount() lifecycle method. @Ogutu-Brian

File: src/views/Pagination/PaginationContainer.jsx (hunk 1 above)
Comment (Amoswachira) [reply src/views/Pagination/PaginationContainer.jsx]: @Ogutu-Brian let me fix that

File: src/views/Articles/Posts.jsx (hunk 2 above)
Comment (Amoswachira) [reply src/views/Articles/Posts.jsx:16]: @Ogutu-Brian  and @IssaIan  solved

File: src/views/Pagination/PaginationContainer.jsx (hunk 1 above)
Comment (Amoswachira) [reply src/views/Pagination/PaginationContainer.jsx]: @Ogutu-Brian fixed

**Review by Ogutu-Brian — APPROVED** [review]

**Pull request merged**